share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -0,0 +1,187 @@
+    if("${var_name}" MATCHES "^${ZBUILD_APPLICATION}_.*")
+      list(APPEND application_vars ${var_name})
+      continue()
+    endif()
+
+    # This means there is a match to another image than current one, ignore.
+    if("${var_name}" MATCHES "^.*_CONFIG_.*")
+      continue()
+    endif()
+
+    # sysbuild reserved namespace.
+    if(var_name IN_LIST sysbuild_vars OR "${var_name}" MATCHES "^SB_CONFIG_.*")
+      continue()
+    endif()
+
+    if("${var_name}" MATCHES "^CONFIG_.*")
+      if(ZBUILD_MAIN_APP)
+        list(APPEND application_vars ${var_name})
+      endif()
+      continue()
+    endif()
+
+    if(var_name IN_LIST shared_image_variables_list)
+      list(APPEND application_vars ${var_name})
+      continue()
+    endif()
+
+    if("${var_name}" MATCHES "${shared_image_variables_regex}")
+      list(APPEND application_vars ${var_name})
+    endif()
+  endforeach()
+
+  foreach(app_var_name ${application_vars})
+    string(REGEX REPLACE "^${ZBUILD_APPLICATION}_" "" var_name "${app_var_name}")
+    get_property(var_type  CACHE ${app_var_name} PROPERTY TYPE)
+    set(new_cache_entry "${var_name}:${var_type}=${${app_var_name}}")
+    if(NOT new_cache_entry IN_LIST app_cache_strings)
+      # This entry does not exists, let's see if it has been updated.
+      foreach(entry ${app_cache_strings})
+        if("${entry}" MATCHES "^${var_name}:.*")
+          list(REMOVE_ITEM app_cache_strings "${entry}")
+          break()
+        endif()
+      endforeach()
+      list(APPEND app_cache_strings "${var_name}:${var_type}=${${app_var_name}}")
+      list(APPEND app_cache_entries "-D${var_name}:${var_type}=${${app_var_name}}")
+    endif()
+  endforeach()
+
+  if(NOT "${app_cache_strings_current}" STREQUAL "${app_cache_strings}")
+    string(REPLACE ";" "\n" app_cache_strings "${app_cache_strings}")
+    file(WRITE ${app_cache_file} ${app_cache_strings})
+  endif()
+
+  if(DEFINED ZBUILD_BOARD)
+    list(APPEND app_cache_entries "-DBOARD=${ZBUILD_BOARD}")
+  elseif(NOT ZBUILD_MAIN_APP)
+    list(APPEND app_cache_entries "-DBOARD=${BOARD}")
+  endif()
+
+  set(image_banner "* Running CMake for ${ZBUILD_APPLICATION} *")
+  string(LENGTH "${image_banner}" image_banner_width)
+  string(REPEAT "*" ${image_banner_width} image_banner_header)
+  message(STATUS "\n   ${image_banner_header}\n"
+                 "   ${image_banner}\n"
+                 "   ${image_banner_header}\n"
+  )
+
+  execute_process(
+    COMMAND ${CMAKE_COMMAND}
+      -G${CMAKE_GENERATOR}
+      ${app_cache_entries}
+      -B${CMAKE_BINARY_DIR}/${ZBUILD_APPLICATION}
+      -S${ZBUILD_SOURCE_DIR}
+    RESULT_VARIABLE   return_val
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+  )
+
+  if(return_val)
+    message(FATAL_ERROR
+            "CMake configure failed for Zephyr project: ${ZBUILD_APPLICATION}\n"
+            "Location: ${ZBUILD_SOURCE_DIR}"
+    )
+  endif()
+
+  foreach(kconfig_target
+      menuconfig
```

Comment:
${KCONFIG_TARGETS} is visible here. Isn't it.

Reply:
it is, but the `KCONFIG_TARGETS` at this point contains the sysbuild targets, not the kconfig targets for the subprojects we are including.
Ref:
https://github.com/zephyrproject-rtos/zephyr/blob/70e05e2d2a379ba71d24430382a183706c865142/share/sysbuild/cmake/modules/sysbuild_kconfig.cmake#L13